ext/spl/spl_directory.stub.php

```diff
@@ -190,101 +190,101 @@ class SplFileObject extends SplFileInfo implements RecursiveIterator, SeekableIt
 
-    /** @return array|false */
-    public function fgetcsv(string $separator = ",", string $enclosure = "\"", string $escape = "\\") {}
+    /** @tentative-return-type */
+    public function fgetcsv(string $separator = ",", string $enclosure = "\"", string $escape = "\\"): array|false|null {}
 
-    /** @return int|false */
-    public function fputcsv(array $fields, string $separator = ",", string $enclosure = "\"", string $escape = "\\", string $eol = "\n") {}
+    /** @tentative-return-type */
+    public function fputcsv(array $fields, string $separator = ",", string $enclosure = "\"", string $escape = "\\", string $eol = "\n"): int|false {}
 
-    /** @return bool|null */
-    public function setCsvControl(string $separator = ",", string $enclosure = "\"", string $escape = "\\") {}
+    /** @tentative-return-type */
+    public function setCsvControl(string $separator = ",", string $enclosure = "\"", string $escape = "\\"): void {}
 
-    /** @return array */
-    public function getCsvControl() {}
+    /** @tentative-return-type */
+    public function getCsvControl(): array {}
 
     /**
      * @param int $wouldBlock
-     * @return bool
+     * @tentative-return-type
      */
-    public function flock(int $operation, &$wouldBlock = null) {}
+    public function flock(int $operation, &$wouldBlock = null): bool {}
 
-    /** @return bool */
-    public function fflush() {}
+    /** @tentative-return-type */
+    public function fflush(): bool {}
 
-    /** @return int|false */
-    public function ftell() {}
+    /** @tentative-return-type */
+    public function ftell(): int|false {}
 
-    /** @return int */
-    public function fseek(int $offset, int $whence = SEEK_SET) {}
+    /** @tentative-return-type */
+    public function fseek(int $offset, int $whence = SEEK_SET): int {}
 
-    /** @return string|false */
-    public function fgetc() {}
+    /** @tentative-return-type */
+    public function fgetc(): string|false {}
 
-    /** @return int */
-    public function fpassthru() {}
+    /** @tentative-return-type */
+    public function fpassthru(): int {}
 
-    /** @return array|int|false|null */
-    public function fscanf(string $format, mixed &...$vars) {}
+    /** @tentative-return-type */
+    public function fscanf(string $format, mixed &...$vars): array|int|null {}
 
-    /** @return int|false */
-    public function fwrite(string $data, int $length = 0) {}
+    /** @tentative-return-type */
+    public function fwrite(string $data, int $length = 0): int|false {}
 
-    /** @return array */
-    public function fstat() {}
+    /** @tentative-return-type */
+    public function fstat(): array {}
 
-    /** @return bool */
-    public function ftruncate(int $size) {}
+    /** @tentative-return-type */
+    public function ftruncate(int $size): bool {}
 
-    /** @return string|array|false */
-    public function current() {}
+    /** @tentative-return-type */
+    public function current(): string|array|false {}
```

Comment:
We might want to try getting rid of `false` here as well.

Reply:
Hm, I think `false` is actually also possible through the current_zval codepath, which is populated by fgetcsv, which can return false. So I guess this has to stay.

Reply:
Looks like it can also be populated by `getCurrentLine()`, which is overridable. Now, this patch adds a tentative `: string` type to it, but that can still be bypassed. We probably need to add an explicit check in the implementation that makes sure it only returns string?

Reply:
This is addressed by https://github.com/php/php-src/commit/8a67dfd16b2032b8e575e8a99cf9a74a1014ec9d.